client/components/stats-date-control/index.tsx

```diff
@@ -66,7 +104,7 @@ const StatsDateControl = ( {
 		const period = bestPeriodForDays( rangeInDays );
 
 		const event_from = isOdysseyStats ? 'jetpack_odyssey' : 'calypso';
-		recordTracksEvent( `${ event_from }_stats_date_picker_apply_button_clicked` );
+		recordTracksEvent( eventNames[ event_from ][ 'apply_button' ] );
```

Comment:
We might also want to make it more robust in case of any missed keys

```JavaScript
eventNames[ event_from ]?.apply_button  && recordTracksEvent( eventNames[ event_from ].apply_button );
```